src/providers/ContentShareProvider/state.tsx

```diff
@@ -11,71 +13,177 @@ export enum ContentActionType {
-type UpdateAction = {
-  type: ContentActionType.UPDATE;
-  payload: UpdatePayload;
+type StartingPayload = EmptyPayload;
+type DidStopPayload = {
+  localAttendeeId: string;
 };
-
 type UpdatePayload = {
   isLocalUser: boolean;
   tileState: VideoTileState;
 };
-
-type TogglePauseAction = {
-  type: ContentActionType.TOGGLE_PAUSE;
-  payload?: any;
-};
-
-type RemoveAction = {
-  type: ContentActionType.REMOVE;
-  payload: number;
-};
-
-type DeniedAction = {
-  type: ContentActionType.DENIED;
-  payload?: any;
+type TogglePausePayload = EmptyPayload;
+type RemovePayload = {
+  tileId: number;
+  isLocalUser: boolean;
 };
-
-type ResetAction = {
-  type: ContentActionType.RESET;
-  payload?: any;
+type DeniedPayload = EmptyPayload;
+type ResetPayload = EmptyPayload;
+type UpdateCanStartPayload = {
+  maxContentShares: number;
 };
 
+// Action types
 export type Action =
-  | StartingAction
-  | DidStopAction
-  | UpdateAction
-  | TogglePauseAction
-  | RemoveAction
-  | DeniedAction
-  | ResetAction;
+  | { type: ContentActionType.STARTING; payload?: StartingPayload }
+  | { type: ContentActionType.DID_STOP; payload: DidStopPayload }
+  | { type: ContentActionType.UPDATE; payload: UpdatePayload }
+  | { type: ContentActionType.TOGGLE_PAUSE; payload?: TogglePausePayload }
+  | { type: ContentActionType.REMOVE; payload: RemovePayload }
+  | { type: ContentActionType.DENIED; payload?: DeniedPayload }
+  | { type: ContentActionType.RESET; payload?: ResetPayload }
+  | {
+      type: ContentActionType.UPDATE_CAN_START;
+      payload: UpdateCanStartPayload;
+    };
 
+/**
+ * ContentShareState represents the state of content sharing in a meeting.
+ * It includes both backward compatibility properties for single content share
+ * and new properties for supporting multiple content shares.
+ */
 export type ContentShareState = {
-  tileId: number | null;
   paused: boolean;
   isLocalShareLoading: boolean;
   isLocalUserSharing: boolean;
+  canStartContentShare: boolean;
+
+  /**
+   * @deprecated Use tiles, tileIdToAttendeeId, and attendeeIdToTileId instead.
+   * This will be removed in a future version.
+   */
+  tileId: number | null;
+
+  /**
+   * @deprecated Use tiles, tileIdToAttendeeId, and attendeeIdToTileId instead.
+   * This will be removed in a future version.
+   */
   sharingAttendeeId: string | null;
+
+  tiles: number[];
+  tileIdToAttendeeId: TileMap;
+  attendeeIdToTileId: AttendeeMap;
 };
 
 export const initialState: ContentShareState = {
-  tileId: null,
   paused: false,
   isLocalUserSharing: false,
   isLocalShareLoading: false,
+  canStartContentShare: true,
+  tileId: null,
   sharingAttendeeId: null,
+  tiles: [],
+  tileIdToAttendeeId: {},
+  attendeeIdToTileId: {},
+};
+
+const addTileToCollections = (
+  state: ContentShareState,
+  tileId: number,
+  attendeeId: string
+): {
+  tiles: number[];
+  tileIdToAttendeeId: TileMap;
+  attendeeIdToTileId: AttendeeMap;
+} => {
+  return {
+    tiles: [...state.tiles, tileId],
```

Comment:
How about we add the new `tileId` at the start? Would it help us in not calculating the max always?

Reply:
That would work too. But I'd prefer existing explicit check because 
1. in JS SDK video tile id is always increasing
    https://github.com/aws/amazon-chime-sdk-js/blob/320ba8446744b756b2deab9d1254704a7f400a3d/src/videotilecontroller/DefaultVideoTileController.ts#L179-L182

2. Most simple approach, eliminate the need to maintain this in data structure. I think the most promising data structure should be LRU. But it's overkill here since we only have 2 content shares and it's already handled in JS SDK.

3. we are using this check today in line 60 below: https://github.com/aws/amazon-chime-sdk-component-library-react/blob/efb7f0f4387a0929a1e2c9813a0afc57df1e2286/src/providers/ContentShareProvider/index.tsx#L57-L64
```

```

Reply:
Cool no issue!